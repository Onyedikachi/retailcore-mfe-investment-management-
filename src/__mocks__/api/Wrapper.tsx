import { MemoryRouter } from "react-router-dom";
import { Provider } from "react-redux";
import { render } from "@testing-library/react";

import { AppStore, RootState, store as S } from "../../config/store";
import { setupListeners } from "@reduxjs/toolkit/dist/query";
import type { PreloadedState } from "@reduxjs/toolkit";
import type { RenderOptions } from "@testing-library/react";
import { PropsWithChildren } from "react";

interface ExtendedRenderOptions extends Omit<RenderOptions, "queries"> {
  preloadedState?: PreloadedState<RootState>;
  store?: AppStore;
  route?: string;
}

interface IWrapper {
  children: React.ReactNode;
}

export function renderWithProviders(
  ui: React.ReactElement | null,
  {
    preloadedState = {},
    store = S(preloadedState),
    route = "/",
    ...renderOptions
  }: ExtendedRenderOptions = {}
) {
  setupListeners(store.dispatch);
  jest.mock("react-router-dom", () => ({
    BrowserRouter: ({ children }) => <div>{children}</div>,
    Routes: ({ children }) => <div>{children}</div>,
    Route: ({ element }) => element,
  }));
  function Wrappers({ children }: PropsWithChildren<IWrapper>): JSX.Element {
    window.history.pushState({}, "Test page", route);
    return <Provider store={store}>{children}</Provider>;
  }
  return { store, ...render(ui, { wrapper: Wrappers, ...renderOptions }) };
}
