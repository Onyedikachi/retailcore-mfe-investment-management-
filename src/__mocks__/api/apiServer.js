// mswServer.js
import { setupServer } from "msw/node";
import { handlers } from "./apiHandler";

// Create the MSW server instance
export const server = setupServer(...handlers);

// Export the server instance for use in tests
// export { server };
