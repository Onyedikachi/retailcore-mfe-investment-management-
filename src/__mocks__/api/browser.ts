// mswServer.js
import { setupWorker } from "msw";
import { handlers } from "./apiHandler";

// Create the MSW server instance
export const server = setupWorker(...handlers);

// Export the server instance for use in tests
// export { server };
