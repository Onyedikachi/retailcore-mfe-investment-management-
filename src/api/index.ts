export * from "./linkApi";
export * from "./authApi";
export * from "./investmentApi";
export * from "./accountApi";
