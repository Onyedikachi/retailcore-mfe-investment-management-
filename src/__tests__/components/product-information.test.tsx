import { fireEvent, render, screen } from "@testing-library/react";
import ProductInformation from "../../components/pages/term-deposit/forms/product-information";
import {renderWithProviders} from "../../__mocks__/api/Wrapper"
import { act } from "react-dom/test-utils";

const fData = {
    productName: "",
    slogan: "",
  description: "",
  startDate: null,
  endDate: null,
  currency: "",
}

jest.mock("react-router-dom", () => ({
    BrowserRouter: ({ children }) => <div>{children}</div>,
    Routes: ({ children }) => <div>{children}</div>,
    Route: ({ element }) => element,
    useNavigate: jest.fn(),
    useParams: jest.fn().mockReturnValue({process: "continue"}),
  }));
  const navigate = jest.fn();

describe("ProductInformation", () => {
    it ("Renders without error", () => {
        const form = renderWithProviders(<ProductInformation proceed={jest.fn()} formData={fData} setDisabled={jest.fn()}/>)
        expect(form).toMatchSnapshot()
    })
    
    it("Changes values", () => {
        renderWithProviders(<ProductInformation proceed={jest.fn()} formData={fData} setDisabled={jest.fn()}/>)
        const input = screen.getByTestId("product-name")
        act(() => {
            fireEvent.change(input, "new Name")
        })
        screen.debug();
    })
})