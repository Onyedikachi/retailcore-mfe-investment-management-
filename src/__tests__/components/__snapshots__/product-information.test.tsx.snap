// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductInformation Renders without error 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        id="productform"
      >
        <div
          class=""
        >
          <div
            class="mb-6 flex flex-col gap-[1px]"
          >
            <div
              class="flex itemx-center gap-2 w-[300px]"
            >
               
              <label
                class=" pt-[10px]  flex text-base font-semibold text-[#636363]"
              >
                Product Name
                 
                <span
                  class="flex"
                >
                   
                  <div>
                    <svg
                      data-testid="red-dot"
                      fill="none"
                      height="5"
                      viewBox="0 0 5 5"
                      width="5"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <circle
                        cx="2.5"
                        cy="2.5"
                        fill="#CF2A2A"
                        r="2.5"
                      />
                    </svg>
                  </div>
                </span>
              </label>
              <div
                class="flex items-end cursor-pointer"
              >
                <div
                  class="flex items-end !bg-white"
                >
                  <button
                    aria-expanded="false"
                    class="bg-transparent p-0"
                    type="button"
                  >
                    <div
                      class="w-[18px] h-[18px] text-[#636363]"
                    >
                      <svg
                        fill="currentColor"
                        height="1em"
                        stroke="currentColor"
                        stroke-width="0"
                        version="1.1"
                        viewBox="0 0 16 16"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M7 4.75c0-0.412 0.338-0.75 0.75-0.75h0.5c0.412 0 0.75 0.338 0.75 0.75v0.5c0 0.412-0.338 0.75-0.75 0.75h-0.5c-0.412 0-0.75-0.338-0.75-0.75v-0.5z"
                        />
                        <path
                          d="M10 12h-4v-1h1v-3h-1v-1h3v4h1z"
                        />
                        <path
                          d="M8 0c-4.418 0-8 3.582-8 8s3.582 8 8 8 8-3.582 8-8-3.582-8-8-8zM8 14.5c-3.59 0-6.5-2.91-6.5-6.5s2.91-6.5 6.5-6.5 6.5 2.91 6.5 6.5-2.91 6.5-6.5 6.5z"
                        />
                      </svg>
                    </div>
                  </button>
                </div>
              </div>
            </div>
            <div
              class="w-full flex flex-col gap-2"
            >
              <div
                class="relative flex items-center max-w-[642px]"
              >
                <input
                  aria-invalid="false"
                  class="placeholder-[#BCBBBB] ring-0 outline-none w-full pt-[10px] pb-[16px] border-b border-[#8F8F8F] pr-[74px] placeholder:text-[#BCBBBB]  "
                  data-testid="product-name"
                  maxlength="50"
                  placeholder="Enter Name"
                  value=""
                />
                <div
                  class="absolute right-0 text-xs text-[#8F8F8F] flex items-center gap-x-[11px]"
                >
                  <span>
                     
                    50
                    /
                    50
                  </span>
                   
                </div>
              </div>
              
            </div>
          </div>
          <div
            class="mb-6 flex flex-col gap-[1px]"
          >
            <label
              class="w-[300px] pt-[10px]  text-base font-semibold text-[#636363]"
            >
              Slogan
            </label>
            <div
              class="w-full flex flex-col gap-2"
            >
              <div
                class="relative flex items-center"
              >
                <input
                  aria-invalid="false"
                  class="placeholder-[#BCBBBB] ring-0 outline-none w-full pt-[10px] pb-[16px] border-b border-[#8F8F8F] pr-[74px] placeholder:text-[#BCBBBB] "
                  data-testid="investment-slogan"
                  maxlength="160"
                  name="slogan"
                  placeholder="Enter a slogan"
                  value=""
                />
                <div
                  class="absolute right-0 text-xs text-[#8F8F8F] flex items-center gap-x-[11px]"
                >
                  <span>
                    160
                    /
                    160
                  </span>
                   
                </div>
              </div>
              
            </div>
          </div>
          <div
            class="mb-6 flex flex-col gap-[13px]"
          >
            <div
              class="flex  gap-2 w-[300px]"
            >
               
              <label
                class=" pt-[10px] flex text-base font-semibold text-[#636363]"
              >
                Product Description
                 
                <span
                  class="flex"
                >
                   
                  <div>
                    <svg
                      data-testid="red-dot"
                      fill="none"
                      height="5"
                      viewBox="0 0 5 5"
                      width="5"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <circle
                        cx="2.5"
                        cy="2.5"
                        fill="#CF2A2A"
                        r="2.5"
                      />
                    </svg>
                  </div>
                </span>
              </label>
            </div>
            <div
              class="w-full flex flex-col gap-2"
            >
              <div
                class="relative"
              >
                <textarea
                  class="min-h-[150px] w-full rounded-md border border-[#8F8F8F] focus:outline-none px-3 py-[11px] placeholder:text-[#BCBBBB] resize-none "
                  data-testid="product-description"
                  maxlength="250"
                  name="description"
                  placeholder="Enter description"
                />
                
                <span
                  class="absolute bottom-4 right-2 text-xs text-[#8F8F8F] flex items-center gap-x-1"
                >
                  <span>
                    0
                  </span>
                  /
                  <span>
                    250
                  </span>
                </span>
              </div>
            </div>
          </div>
          <div
            class="flex gap-12"
          >
            <div
              class="flex flex-col gap"
            >
              <div
                class="flex  gap-x-2 w-[300px]"
              >
                 
                <label
                  class=" pt-[10px]  text-base font-semibold text-[#636363]"
                >
                  Product Life Cycle
                </label>
                <div
                  class="flex items-end cursor-pointer"
                >
                  <div
                    class="flex items-end !bg-white"
                  >
                    <button
                      aria-expanded="false"
                      class="bg-transparent p-0"
                      type="button"
                    >
                      <div
                        class="w-[18px] h-[18px] text-[#636363]"
                      >
                        <svg
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          version="1.1"
                          viewBox="0 0 16 16"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M7 4.75c0-0.412 0.338-0.75 0.75-0.75h0.5c0.412 0 0.75 0.338 0.75 0.75v0.5c0 0.412-0.338 0.75-0.75 0.75h-0.5c-0.412 0-0.75-0.338-0.75-0.75v-0.5z"
                          />
                          <path
                            d="M10 12h-4v-1h1v-3h-1v-1h3v4h1z"
                          />
                          <path
                            d="M8 0c-4.418 0-8 3.582-8 8s3.582 8 8 8 8-3.582 8-8-3.582-8-8-8zM8 14.5c-3.59 0-6.5-2.91-6.5-6.5s2.91-6.5 6.5-6.5 6.5 2.91 6.5 6.5-2.91 6.5-6.5 6.5z"
                          />
                        </svg>
                      </div>
                    </button>
                  </div>
                </div>
              </div>
              <div
                class="flex gap-x-4"
              >
                <div>
                  <div
                    class="relative flex items-center date-picker"
                  >
                    <div
                      class="react-datepicker-wrapper"
                    >
                      <div
                        class="react-datepicker__input-container react-datepicker__view-calendar-icon"
                      >
                        <svg
                          class="react-datepicker__calendar-icon"
                          viewBox="0 0 448 512"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M96 32V64H48C21.5 64 0 85.5 0 112v48H448V112c0-26.5-21.5-48-48-48H352V32c0-17.7-14.3-32-32-32s-32 14.3-32 32V64H160V32c0-17.7-14.3-32-32-32S96 14.3 96 32zM448 192H0V464c0 26.5 21.5 48 48 48H400c26.5 0 48-21.5 48-48V192z"
                          />
                        </svg>
                        <input
                          class="placeholder-[#BCBBBB] ring-0 outline-none w-full py-1 pl-2 pr-4 border-b placeholder:text-[#BCBBBB] border-[#8F8F8F]"
                          placeholder="dd/mm/yyyy"
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                    <span
                      class="absolute right-[2px]"
                    >
                       
                      <svg
                        fill="currentColor"
                        height="1em"
                        stroke="currentColor"
                        stroke-width="0"
                        viewBox="0 0 448 512"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M12 192h424c6.6 0 12 5.4 12 12v260c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V204c0-6.6 5.4-12 12-12zm436-44v-36c0-26.5-21.5-48-48-48h-48V12c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v52H160V12c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v52H48C21.5 64 0 85.5 0 112v36c0 6.6 5.4 12 12 12h424c6.6 0 12-5.4 12-12z"
                        />
                      </svg>
                    </span>
                  </div>
                </div>
                -
                <div>
                  <div
                    class="relative flex items-center date-picker"
                  >
                    <div
                      class="react-datepicker-wrapper"
                    >
                      <div
                        class="react-datepicker__input-container react-datepicker__view-calendar-icon"
                      >
                        <svg
                          class="react-datepicker__calendar-icon"
                          viewBox="0 0 448 512"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M96 32V64H48C21.5 64 0 85.5 0 112v48H448V112c0-26.5-21.5-48-48-48H352V32c0-17.7-14.3-32-32-32s-32 14.3-32 32V64H160V32c0-17.7-14.3-32-32-32S96 14.3 96 32zM448 192H0V464c0 26.5 21.5 48 48 48H400c26.5 0 48-21.5 48-48V192z"
                          />
                        </svg>
                        <input
                          class="placeholder-[#BCBBBB] ring-0 outline-none w-full py-1 pl-2 pr-4 border-b placeholder:text-[#BCBBBB] border-[#8F8F8F]"
                          placeholder="dd/mm/yyyy"
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                    <span
                      class="absolute right-[2px]"
                    >
                       
                      <svg
                        fill="currentColor"
                        height="1em"
                        stroke="currentColor"
                        stroke-width="0"
                        viewBox="0 0 448 512"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M12 192h424c6.6 0 12 5.4 12 12v260c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V204c0-6.6 5.4-12 12-12zm436-44v-36c0-26.5-21.5-48-48-48h-48V12c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v52H160V12c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v52H48C21.5 64 0 85.5 0 112v36c0 6.6 5.4 12 12 12h424c6.6 0 12-5.4 12-12z"
                        />
                      </svg>
                    </span>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="flex items-end gap"
            >
              <div
                class="w-[300px]"
              >
                <div
                  class=" min-w-full"
                  role="combobox"
                >
                  <div
                    class="flex  gap-2 min-w-[300px]"
                  >
                     
                    <label
                      class="flex  text-base font-semibold text-[#636363]"
                    >
                      Product Currency
                      <div>
                        <svg
                          data-testid="red-dot"
                          fill="none"
                          height="5"
                          viewBox="0 0 5 5"
                          width="5"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <circle
                            cx="2.5"
                            cy="2.5"
                            fill="#CF2A2A"
                            r="2.5"
                          />
                        </svg>
                      </div>
                    </label>
                    <div
                      class="flex items-end cursor-pointer"
                    >
                      <div
                        class="flex items-end !bg-white"
                      >
                        <button
                          aria-expanded="false"
                          class="bg-transparent p-0"
                          type="button"
                        >
                          <div
                            class="w-[18px] h-[18px] text-[#636363]"
                          >
                            <svg
                              fill="currentColor"
                              height="1em"
                              stroke="currentColor"
                              stroke-width="0"
                              version="1.1"
                              viewBox="0 0 16 16"
                              width="1em"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <path
                                d="M7 4.75c0-0.412 0.338-0.75 0.75-0.75h0.5c0.412 0 0.75 0.338 0.75 0.75v0.5c0 0.412-0.338 0.75-0.75 0.75h-0.5c-0.412 0-0.75-0.338-0.75-0.75v-0.5z"
                              />
                              <path
                                d="M10 12h-4v-1h1v-3h-1v-1h3v4h1z"
                              />
                              <path
                                d="M8 0c-4.418 0-8 3.582-8 8s3.582 8 8 8 8-3.582 8-8-3.582-8-8-8zM8 14.5c-3.59 0-6.5-2.91-6.5-6.5s2.91-6.5 6.5-6.5 6.5 2.91 6.5 6.5-2.91 6.5-6.5 6.5z"
                              />
                            </svg>
                          </div>
                        </button>
                      </div>
                    </div>
                  </div>
                  <div
                    class="relative mt-1"
                    data-headlessui-state=""
                  >
                    <button
                      aria-expanded="false"
                      aria-haspopup="listbox"
                      class="relative w-full cursor-pointer  bg-white py-1  pr-10 text-left  border-b border-[#636363] focus:outline-none  text-[#252C32] text-sm border-[#8F8F8F]"
                      data-headlessui-state=""
                      id="headlessui-listbox-button-1"
                      type="button"
                    >
                      <span
                        class="block truncate"
                      >
                        <span
                          class="text-[#aaa]"
                        >
                          Select currency
                        </span>
                      </span>
                      <span
                        class="pointer-events-none absolute inset-y-0 right-0 flex items-center pr-2 text-xs"
                      >
                        <svg
                          fill="none"
                          height="10"
                          viewBox="0 0 15 10"
                          width="15"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M15 0H0L7.14286 9.47368L15 0Z"
                            fill="#636363"
                          />
                        </svg>
                      </span>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      id="productform"
    >
      <div
        class=""
      >
        <div
          class="mb-6 flex flex-col gap-[1px]"
        >
          <div
            class="flex itemx-center gap-2 w-[300px]"
          >
             
            <label
              class=" pt-[10px]  flex text-base font-semibold text-[#636363]"
            >
              Product Name
               
              <span
                class="flex"
              >
                 
                <div>
                  <svg
                    data-testid="red-dot"
                    fill="none"
                    height="5"
                    viewBox="0 0 5 5"
                    width="5"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <circle
                      cx="2.5"
                      cy="2.5"
                      fill="#CF2A2A"
                      r="2.5"
                    />
                  </svg>
                </div>
              </span>
            </label>
            <div
              class="flex items-end cursor-pointer"
            >
              <div
                class="flex items-end !bg-white"
              >
                <button
                  aria-expanded="false"
                  class="bg-transparent p-0"
                  type="button"
                >
                  <div
                    class="w-[18px] h-[18px] text-[#636363]"
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      version="1.1"
                      viewBox="0 0 16 16"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M7 4.75c0-0.412 0.338-0.75 0.75-0.75h0.5c0.412 0 0.75 0.338 0.75 0.75v0.5c0 0.412-0.338 0.75-0.75 0.75h-0.5c-0.412 0-0.75-0.338-0.75-0.75v-0.5z"
                      />
                      <path
                        d="M10 12h-4v-1h1v-3h-1v-1h3v4h1z"
                      />
                      <path
                        d="M8 0c-4.418 0-8 3.582-8 8s3.582 8 8 8 8-3.582 8-8-3.582-8-8-8zM8 14.5c-3.59 0-6.5-2.91-6.5-6.5s2.91-6.5 6.5-6.5 6.5 2.91 6.5 6.5-2.91 6.5-6.5 6.5z"
                      />
                    </svg>
                  </div>
                </button>
              </div>
            </div>
          </div>
          <div
            class="w-full flex flex-col gap-2"
          >
            <div
              class="relative flex items-center max-w-[642px]"
            >
              <input
                aria-invalid="false"
                class="placeholder-[#BCBBBB] ring-0 outline-none w-full pt-[10px] pb-[16px] border-b border-[#8F8F8F] pr-[74px] placeholder:text-[#BCBBBB]  "
                data-testid="product-name"
                maxlength="50"
                placeholder="Enter Name"
                value=""
              />
              <div
                class="absolute right-0 text-xs text-[#8F8F8F] flex items-center gap-x-[11px]"
              >
                <span>
                   
                  50
                  /
                  50
                </span>
                 
              </div>
            </div>
            
          </div>
        </div>
        <div
          class="mb-6 flex flex-col gap-[1px]"
        >
          <label
            class="w-[300px] pt-[10px]  text-base font-semibold text-[#636363]"
          >
            Slogan
          </label>
          <div
            class="w-full flex flex-col gap-2"
          >
            <div
              class="relative flex items-center"
            >
              <input
                aria-invalid="false"
                class="placeholder-[#BCBBBB] ring-0 outline-none w-full pt-[10px] pb-[16px] border-b border-[#8F8F8F] pr-[74px] placeholder:text-[#BCBBBB] "
                data-testid="investment-slogan"
                maxlength="160"
                name="slogan"
                placeholder="Enter a slogan"
                value=""
              />
              <div
                class="absolute right-0 text-xs text-[#8F8F8F] flex items-center gap-x-[11px]"
              >
                <span>
                  160
                  /
                  160
                </span>
                 
              </div>
            </div>
            
          </div>
        </div>
        <div
          class="mb-6 flex flex-col gap-[13px]"
        >
          <div
            class="flex  gap-2 w-[300px]"
          >
             
            <label
              class=" pt-[10px] flex text-base font-semibold text-[#636363]"
            >
              Product Description
               
              <span
                class="flex"
              >
                 
                <div>
                  <svg
                    data-testid="red-dot"
                    fill="none"
                    height="5"
                    viewBox="0 0 5 5"
                    width="5"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <circle
                      cx="2.5"
                      cy="2.5"
                      fill="#CF2A2A"
                      r="2.5"
                    />
                  </svg>
                </div>
              </span>
            </label>
          </div>
          <div
            class="w-full flex flex-col gap-2"
          >
            <div
              class="relative"
            >
              <textarea
                class="min-h-[150px] w-full rounded-md border border-[#8F8F8F] focus:outline-none px-3 py-[11px] placeholder:text-[#BCBBBB] resize-none "
                data-testid="product-description"
                maxlength="250"
                name="description"
                placeholder="Enter description"
              />
              
              <span
                class="absolute bottom-4 right-2 text-xs text-[#8F8F8F] flex items-center gap-x-1"
              >
                <span>
                  0
                </span>
                /
                <span>
                  250
                </span>
              </span>
            </div>
          </div>
        </div>
        <div
          class="flex gap-12"
        >
          <div
            class="flex flex-col gap"
          >
            <div
              class="flex  gap-x-2 w-[300px]"
            >
               
              <label
                class=" pt-[10px]  text-base font-semibold text-[#636363]"
              >
                Product Life Cycle
              </label>
              <div
                class="flex items-end cursor-pointer"
              >
                <div
                  class="flex items-end !bg-white"
                >
                  <button
                    aria-expanded="false"
                    class="bg-transparent p-0"
                    type="button"
                  >
                    <div
                      class="w-[18px] h-[18px] text-[#636363]"
                    >
                      <svg
                        fill="currentColor"
                        height="1em"
                        stroke="currentColor"
                        stroke-width="0"
                        version="1.1"
                        viewBox="0 0 16 16"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M7 4.75c0-0.412 0.338-0.75 0.75-0.75h0.5c0.412 0 0.75 0.338 0.75 0.75v0.5c0 0.412-0.338 0.75-0.75 0.75h-0.5c-0.412 0-0.75-0.338-0.75-0.75v-0.5z"
                        />
                        <path
                          d="M10 12h-4v-1h1v-3h-1v-1h3v4h1z"
                        />
                        <path
                          d="M8 0c-4.418 0-8 3.582-8 8s3.582 8 8 8 8-3.582 8-8-3.582-8-8-8zM8 14.5c-3.59 0-6.5-2.91-6.5-6.5s2.91-6.5 6.5-6.5 6.5 2.91 6.5 6.5-2.91 6.5-6.5 6.5z"
                        />
                      </svg>
                    </div>
                  </button>
                </div>
              </div>
            </div>
            <div
              class="flex gap-x-4"
            >
              <div>
                <div
                  class="relative flex items-center date-picker"
                >
                  <div
                    class="react-datepicker-wrapper"
                  >
                    <div
                      class="react-datepicker__input-container react-datepicker__view-calendar-icon"
                    >
                      <svg
                        class="react-datepicker__calendar-icon"
                        viewBox="0 0 448 512"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M96 32V64H48C21.5 64 0 85.5 0 112v48H448V112c0-26.5-21.5-48-48-48H352V32c0-17.7-14.3-32-32-32s-32 14.3-32 32V64H160V32c0-17.7-14.3-32-32-32S96 14.3 96 32zM448 192H0V464c0 26.5 21.5 48 48 48H400c26.5 0 48-21.5 48-48V192z"
                        />
                      </svg>
                      <input
                        class="placeholder-[#BCBBBB] ring-0 outline-none w-full py-1 pl-2 pr-4 border-b placeholder:text-[#BCBBBB] border-[#8F8F8F]"
                        placeholder="dd/mm/yyyy"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                  <span
                    class="absolute right-[2px]"
                  >
                     
                    <svg
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 448 512"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M12 192h424c6.6 0 12 5.4 12 12v260c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V204c0-6.6 5.4-12 12-12zm436-44v-36c0-26.5-21.5-48-48-48h-48V12c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v52H160V12c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v52H48C21.5 64 0 85.5 0 112v36c0 6.6 5.4 12 12 12h424c6.6 0 12-5.4 12-12z"
                      />
                    </svg>
                  </span>
                </div>
              </div>
              -
              <div>
                <div
                  class="relative flex items-center date-picker"
                >
                  <div
                    class="react-datepicker-wrapper"
                  >
                    <div
                      class="react-datepicker__input-container react-datepicker__view-calendar-icon"
                    >
                      <svg
                        class="react-datepicker__calendar-icon"
                        viewBox="0 0 448 512"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M96 32V64H48C21.5 64 0 85.5 0 112v48H448V112c0-26.5-21.5-48-48-48H352V32c0-17.7-14.3-32-32-32s-32 14.3-32 32V64H160V32c0-17.7-14.3-32-32-32S96 14.3 96 32zM448 192H0V464c0 26.5 21.5 48 48 48H400c26.5 0 48-21.5 48-48V192z"
                        />
                      </svg>
                      <input
                        class="placeholder-[#BCBBBB] ring-0 outline-none w-full py-1 pl-2 pr-4 border-b placeholder:text-[#BCBBBB] border-[#8F8F8F]"
                        placeholder="dd/mm/yyyy"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                  <span
                    class="absolute right-[2px]"
                  >
                     
                    <svg
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 448 512"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M12 192h424c6.6 0 12 5.4 12 12v260c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V204c0-6.6 5.4-12 12-12zm436-44v-36c0-26.5-21.5-48-48-48h-48V12c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v52H160V12c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v52H48C21.5 64 0 85.5 0 112v36c0 6.6 5.4 12 12 12h424c6.6 0 12-5.4 12-12z"
                      />
                    </svg>
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="flex items-end gap"
          >
            <div
              class="w-[300px]"
            >
              <div
                class=" min-w-full"
                role="combobox"
              >
                <div
                  class="flex  gap-2 min-w-[300px]"
                >
                   
                  <label
                    class="flex  text-base font-semibold text-[#636363]"
                  >
                    Product Currency
                    <div>
                      <svg
                        data-testid="red-dot"
                        fill="none"
                        height="5"
                        viewBox="0 0 5 5"
                        width="5"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <circle
                          cx="2.5"
                          cy="2.5"
                          fill="#CF2A2A"
                          r="2.5"
                        />
                      </svg>
                    </div>
                  </label>
                  <div
                    class="flex items-end cursor-pointer"
                  >
                    <div
                      class="flex items-end !bg-white"
                    >
                      <button
                        aria-expanded="false"
                        class="bg-transparent p-0"
                        type="button"
                      >
                        <div
                          class="w-[18px] h-[18px] text-[#636363]"
                        >
                          <svg
                            fill="currentColor"
                            height="1em"
                            stroke="currentColor"
                            stroke-width="0"
                            version="1.1"
                            viewBox="0 0 16 16"
                            width="1em"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M7 4.75c0-0.412 0.338-0.75 0.75-0.75h0.5c0.412 0 0.75 0.338 0.75 0.75v0.5c0 0.412-0.338 0.75-0.75 0.75h-0.5c-0.412 0-0.75-0.338-0.75-0.75v-0.5z"
                            />
                            <path
                              d="M10 12h-4v-1h1v-3h-1v-1h3v4h1z"
                            />
                            <path
                              d="M8 0c-4.418 0-8 3.582-8 8s3.582 8 8 8 8-3.582 8-8-3.582-8-8-8zM8 14.5c-3.59 0-6.5-2.91-6.5-6.5s2.91-6.5 6.5-6.5 6.5 2.91 6.5 6.5-2.91 6.5-6.5 6.5z"
                            />
                          </svg>
                        </div>
                      </button>
                    </div>
                  </div>
                </div>
                <div
                  class="relative mt-1"
                  data-headlessui-state=""
                >
                  <button
                    aria-expanded="false"
                    aria-haspopup="listbox"
                    class="relative w-full cursor-pointer  bg-white py-1  pr-10 text-left  border-b border-[#636363] focus:outline-none  text-[#252C32] text-sm border-[#8F8F8F]"
                    data-headlessui-state=""
                    id="headlessui-listbox-button-1"
                    type="button"
                  >
                    <span
                      class="block truncate"
                    >
                      <span
                        class="text-[#aaa]"
                      >
                        Select currency
                      </span>
                    </span>
                    <span
                      class="pointer-events-none absolute inset-y-0 right-0 flex items-center pr-2 text-xs"
                    >
                      <svg
                        fill="none"
                        height="10"
                        viewBox="0 0 15 10"
                        width="15"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M15 0H0L7.14286 9.47368L15 0Z"
                          fill="#636363"
                        />
                      </svg>
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
